### DISCUSSION TITLE
How to execute `xmake install` during `xmake` command?

### DISCUSSION THREAD

**Discussion by yh-sb** [xmake-io/xmake#5352] [Q&A] [answered]
I am using xmake for Qt project. Qt requires to keep its .dll libraries together with built .exe to find them in runtime.
So, even for development (debug or testing), I need to execute `xmake install -o build/install` each time after `xmake` call.

I want to automate this in the way that `xmake install` is called automatically in the end of `xmake` call. So, need to combine both these calls to one `xmake` command.
**How to do this? Is this common practice?**

**Comment by waruqi** [comment, marked as answer]
> So, even for development (debug or testing), I need to execute xmake install -o build/install each time after xmake call.

you should use `xmake run` to run and debug it. it will load all dll files automatically.

**Reply by yh-sb** [reply]
Oh, thank you.
How exactly `xmake run` command works? Does it modify local PATH env. or LIB search path before starting target executable?
I see no extra output with `-v` option.

+ I also found that `xmake run` suppress all stdout output. **Is this expected?** NOTE: When I run `xmake run -d` I see my stdout output from Qt app.

**Reply by waruqi** [reply]
it will bind all so lib path to `LD_LIBRARY_PATH`. and `xmake run` call `os.execv`. it will not capture any stdout.

https://github.com/xmake-io/xmake/blob/2e0fb34b2cd7824e9a4a92f307c33feea5634862/xmake/actions/run/main.lua#L57
